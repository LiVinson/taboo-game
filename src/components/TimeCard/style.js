import styled from 'styled-components'

//Container for the progress bar. Will be the background as the width of the inner bar decreases
export const ProgressBar = styled.div`
	width: 100%;
	position: relative;
	background-color: ${(props) => props.theme.color.grayDark1};
	height: 3rem;
	margin-bottom: 1rem;
	border-radius: 3px;
	border: solid 2px ${(props) => props.theme.color.grayDark2};
	box-shadow: 0px 3px 2px rgba(0, 0, 0, 0.3);
	display: flex;
	justify-content: space-between;
`

//Serves as the progress bar. Width is based on a percentage of time remaining / 60 seconds
export const Bar = styled.div`
	/* allows the color bar to overlap the numbers */
	position: absolute;
	top: 0;
	left: 0;
	width: ${(props) => props.width + '%'};
	/* If width is less than 16 (10 seconds or less remaining): change color */
	background-color: ${(props) =>
		parseInt(props.width) > 16 ? props.theme.color.accent3 : props.theme.color.accent2};
	height: 100%;
	z-index: 2;
`
export const TimeText = styled.div`
	/* positions numbers to the far right */
	position: absolute;
	/* when time = 0, centers time up message */
    right: ${(props) => (parseInt(props.width) > 0 ? 0 : '50%')};
	transform: ${(props) => {
		if (parseInt(props.width) > 0) {
			return 'translateX(0)'
		} else {
			return 'translateX(50%)'
		}
	}};
	padding: 0 0.5rem;
	height: 100%;
	font-size: 1.8rem;
	z-index: 5;
	color: ${(props) => props.theme.color.lightGray1};
`
