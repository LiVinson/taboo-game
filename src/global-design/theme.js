const theme = {
  color: {
    primary: "#1982C4", //light blue
    accent1: "#8AC926", //lime green
    accent2: "#ee5b2c", //orange
    accent3: "#a23b72", //purple
    primaryLight: "#58b1e9", //light-blue
    white: "#fff",
    offWhite: "#f7f9f9",
    lightGray1: "#f1f1f1",
    lightGray2: "#ebebeb",
    grayDark1: "#757575",
    grayDark2: "#333333"
  },
  text: {
    display: "'Oswald', sans-serif",
  },
  fontSize: {
    xsmall: "1.8rem",
    small: "2.2rem",
    medium: "2.7rem",
    large: "4rem",
    xlarge: "7rem",
  },
}


export default theme
