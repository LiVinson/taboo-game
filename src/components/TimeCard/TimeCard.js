import React from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'
import { ProgressBar, Bar, TimeText } from './style'

class TimeCard extends React.Component {
	constructor(props) {
		super(props)
		this.state = {
			timeRemaining: 60,
		}
	}

	//Calculates the difference between current time and time round ends in seconds
	//Updates state with the amount of time remaining every second until 0, then giver
	//calld method to endRound and change to PostRound.
	componentDidMount() {
		const { roundEndTime } = this.props
		let timeRemaining = moment(roundEndTime).diff(moment(), 'second')
		this.intervalId = setInterval(() => {
			if (timeRemaining >= 0) {
				timeRemaining--
				this.setState({
					timeRemaining,
				})
			} else {
				clearInterval(this.intervalId)
				//Only one player needs to update the round for everyone
				if (this.props.role === 'giver') {
					this.props.endRound()
				}
			}
		}, 1000)
	}

	componentWillUnmount() {
		if (this.intervalId) {
			clearInterval(this.intervalId)
		}
	}
	
	render() {
		const width = Math.round((this.state.timeRemaining / 60) * 100)
		return (
			<ProgressBar>
				<Bar width={width}></Bar>
				<TimeText width={width}>
					{this.state.timeRemaining > 0 ? this.state.timeRemaining : "Time's Up!"}
				</TimeText>
			</ProgressBar>
		)
	}
}

TimeCard.propTypes = {
	roundEndTime: PropTypes.number,
	role: PropTypes.string.isRequired,
	endRound: PropTypes.func.isRequired,
}
export default TimeCard
