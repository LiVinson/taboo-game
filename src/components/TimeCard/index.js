import TimeCard from "./TimeCard"

export default TimeCard