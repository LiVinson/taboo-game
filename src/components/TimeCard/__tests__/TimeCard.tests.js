import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from 'styled-components'
import theme from '../../../global-design/theme'
import TimeCard from "../TimeCard"

test("TimeCard renders correctly", () => {
    const wrapper = renderer.create(
        <ThemeProvider theme={theme}>
            <TimeCard />
            </ThemeProvider>
        ).toJSON()
        
    expect(wrapper).toMatchSnapshot()
})