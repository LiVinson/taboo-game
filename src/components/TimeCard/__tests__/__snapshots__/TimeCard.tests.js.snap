// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeCard rendering and functionality TimeCard renders correctly 1`] = `
<div
  className="sc-AxjAm gtWhgV"
>
  <div
    className="sc-AxirZ hQvBzJ"
    width={100}
  />
  <div
    className="sc-AxiKw hqcVBg"
    width={100}
  >
    60
  </div>
</div>
`;
